### PR TITLE
Improve docs for enabling extensions

### LINKED ISSUE #4968: Improve docs for enabling extensions
# Improve documentation

## Link 

[Extensions](https://supabase.com/docs/guides/database/extensions)

## Describe the problem

The documentation provides an example of how to enable extensions via SQL: ```create extension pgtap;```

However, there seem to be 2 issues with this example:
 - In order to ensure that the extension is installed in the ```extensions``` schema, the statement should be ```create extension pgtap with schema extensions```;
 - Enabling some extensions using the statement above leads to permission issues (e.g. ```dblink```, ```http```, ```pg_cron```). More specifically:
    - The extensions CANNOT be enabled with the ```postgres``` users, because this leads to the following error:```
Query 1: ERROR:  permission denied to create extension "plpgsql_check"
HINT:  Must be superuser to create this extension.```

    - If the extensions are enabled via the Dashboard query editor (which uses the supabase_admin role), the enabling succeeds, however, other roles (e.g. ```postgres```) cannot use the extensions;
    - NOTE: this does NOT seem to be a problem with all extensions. For example ```pgtap``` can be installed with the ```postgres``` user (which, unlike the extensions listed above as problematic, has NO native code as far as I know);

## Describe the improvement

I think there need to be 2 improvements addressing both issues above:
- specify the extensions schema in the example;
- document which users to use when enabling extensions OR specify how to enable extensions with supabase_admin and still allow other roles to have access to extensions;

### PR DESCRIPTION
**Pull request by gitstart**
**What kind of change does this PR introduce?**
Partially fixes #4968

One of the postgres extension supported by Supabase , pgcron is introduced in the docs of extension. Here the usage, enabling of extension, functions and methods alongside examples that can be used with pgcron extension are introduced .

**What is the current behavior?**
Issue https://github.com/GitStartHQ/client-supabase/issues/28

175004950-fa2a74bd-886d-47a1-8ff1-99f85e7304b8

**What is the new behavior?**
Added a separate docs section for pgcron extension.

screencapture-localhost-3006-docs-guides-database-extensions-pgcron-2022-06-23-17_18_45

**Additional context**
**To Do**

-  ensure that the extension is installed
-  enabling some extensions may lead to permission issues

### CODE REVIEW

File: web/docs/guides/database/extensions/pg-cron.mdx [hunk 1]
```diff
@@ -0,0 +1,109 @@
+---
+id: pgcron
+title: "pg_cron: Postgres Schedular"
+description: An HTTP Client for PostgreSQL Functions.
```
Comment (kiwicopple) [inline web/docs/guides/database/extensions/pg-cron.mdx:4]: This one might need fixing!

File: web/docs/guides/database/extensions/pg-cron.mdx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+The `pg-cron` extension allows you to schedule job for PostgreSQL.
```
Comment (kiwicopple) [inline web/docs/guides/database/extensions/pg-cron.mdx:10]:
```suggestion
The `pg_cron` extension allows you to schedule job for PostgreSQL.
```

File: web/docs/guides/database/extensions/pg-cron.mdx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+## Overview
+
+Let's cover some basic concepts:
+
+- pg_corn:pg_cron is a simple cron-based job scheduler for PostgreSQL (10 or higher) that runs inside the database as an extension.
```
Comment (kiwicopple) [inline web/docs/guides/database/extensions/pg-cron.mdx:15]:
```suggestion
- pg_cron is a simple cron-based job scheduler for PostgreSQL (10 or higher) that runs inside the database as an extension.
```

File: web/docs/guides/database/extensions/pg-cron.mdx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  - `schedule` − Schedules a job to be executed during a certain span of time.
+  - `unschedule` − Stops a scheduling of a job.
+  - `alter_job` − Function to change job properties.
+  - `schedule_in_database` - Function to schedule in a custom database.
+  - `log_min_messages` - Setting to control log_min_messages in pg_corn launcher.
+  - `enable_superuser_jobs` - Setting to disallow superuser jobs.
+
+You can use the `pg_cron` extension to make these network requests from Postgres.
```
Comment (kiwicopple) [inline web/docs/guides/database/extensions/pg-cron.mdx:23]: I don't know about the "network requests" - it's really just for managing cron jobs, and has nothing to do with networking

File: web/docs/guides/database/extensions/pg-cron.mdx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+## Overview
+
+Let's cover some basic concepts:
+
+- pg_corn:pg_cron is a simple cron-based job scheduler for PostgreSQL (10 or higher) that runs inside the database as an extension.
+  - `schedule` − Schedules a job to be executed during a certain span of time.
+  - `unschedule` − Stops a scheduling of a job.
+  - `alter_job` − Function to change job properties.
+  - `schedule_in_database` - Function to schedule in a custom database.
+  - `log_min_messages` - Setting to control log_min_messages in pg_corn launcher.
```
Comment (kiwicopple) [inline web/docs/guides/database/extensions/pg-cron.mdx:20]:
```suggestion
  - `log_min_messages` - Setting to control log_min_messages in pg_cron launcher.
```

File: web/docs/guides/database/extensions/pg-cron.mdx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+It's good practice to create the extension within a separate schema (like `extensions`) to keep your database clean.
+
+</TabItem>
+
+</Tabs>
+
+### Available functions
+
+While the main usage is simply `pg_cron` enabling cron jobs in postgres , there are different functions for specific functionality:
```
Comment (kiwicopple) [inline web/docs/guides/database/extensions/pg-cron.mdx:71]:
```suggestion
While the main usage is simply enabling cron jobs in Postgres, there are different functions for specific functionality:
```

File: web/docs/guides/database/extensions/pg-cron.mdx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+- `cron.schedule()` 
+- `cron.unschedule()` 
+- `cron.alter_job()`
+- `schedule_in_database`
+- `log_min_messages`
+- `enable_superuser_jobs`
```
Comment (kiwicopple) [inline web/docs/guides/database/extensions/pg-cron.mdx:78]: Will we add more details for each one of these?

Comment (gitstart) [reply web/docs/guides/database/extensions/pg-cron.mdx:78]: @kiwicopple We've extensive docs on the official github repository for `pg_cron`, we've added a reference to the docs at the bottom of this page

File: web/docs/guides/database/extensions/pg-cron.mdx [hunk 8]
```diff
@@ -0,0 +1,109 @@
+---
+id: pgcron
+title: "pg_cron: Postgres Schedular"
```
Comment (kiwicopple) [inline web/docs/guides/database/extensions/pg-cron.mdx:3]: spelling error in scheduler

Comment (kiwicopple) [reply web/docs/guides/database/extensions/pg-cron.mdx:3]: (needs to be fixed everywhere)

Comment (gitstart) [reply web/docs/guides/database/extensions/pg-cron.mdx]: @kiwicopple All grammar errors have been fixed

**Comment by gitstart** [comment]
@kiwicopple Any update on this?

**Pull request closed without merging**